lib/pigeon/dispatcher_worker.ex

```diff
@@ -60,28 +63,26 @@ defmodule Pigeon.DispatcherWorker do
   end
 
   @impl GenServer
-  def handle_info({:"$push", notification}, %{adapter: adapter, state: state}) do
-    case adapter.handle_push(notification, state) do
-      {:noreply, new_state} ->
-        {:noreply, %{adapter: adapter, state: new_state}}
-
-      {:stop, reason, new_state} ->
-        {:stop, reason, %{adapter: adapter, state: new_state}}
+  def handle_info({:"$push", notification}, state) do
+    case state.adapter.handle_push(notification, state) do
+      {:noreply, new_state} -> {:noreply, new_state}
+      {:stop, reason, new_state} -> {:stop, reason, new_state}
     end
   end
 
-  def handle_info(msg, %{adapter: adapter, state: state}) do
-    case adapter.handle_info(msg, state) do
-      {:noreply, new_state} ->
-        {:noreply, %{adapter: adapter, state: new_state}}
+  def handle_info({:set_penalty_box, value, duration_ms}, state) do
+    {:noreply, set_penalty_box(state, value, duration_ms)}
+  end
 
-      {:stop, reason, new_state} ->
-        {:stop, reason, %{adapter: adapter, state: new_state}}
+  def handle_info(msg, state) do
+    case state.adapter.handle_info(msg, state) do
+      {:noreply, new_state} -> {:noreply, new_state}
+      {:stop, reason, new_state} -> {:stop, reason, new_state}
     end
   end
 
   @impl GenServer
-  def handle_call(:info, _from, %{adapter: adapter, state: state}) do
```

Comment:
since you changed state here from the specific adapter state to just the wrapping worker state (including adapter), should we update the peername call to fetch state.state instead? Seems like you changed what this was and didn't update all the calls to it.

Reply:
I changed `init` too so all the properties are now just directly on `state`. There's no more `state.state`.